src/parser.rs

```diff
@@ -3,9 +3,59 @@ use error::Error;
 
 use nom::types::CompleteStr;
 use nom::{eol, space, space0};
+use nom_locate::LocatedSpan;
 
+use std::fmt;
 use std::num;
 
+pub type Span<'a> = LocatedSpan<CompleteStr<'a>>;
+
+pub enum ErrorCode {
+    ExpectedStringExpression,
+    Unknown,
+}
+
+impl From<u32> for ErrorCode {
+    fn from(code: u32) -> Self {
+        match code {
+            0 => ErrorCode::ExpectedStringExpression,
+            _ => ErrorCode::Unknown,
+        }
+    }
+}
+
+impl fmt::Display for ErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ErrorCode::ExpectedStringExpression => f.write_str("string expression expected"),
```

Comment:
Is `string expression expected` from the ECMA script?

Reply:
I am not sure. It is from the error output of the test though.